app/models/qa/iri_template/variable_map.rb

```diff
@@ -30,9 +32,8 @@ def required?
       # @returns the value to use (e.g. 'fr')
       def simple_value(sub_value = nil)
         return sub_value.to_s if sub_value.present?
-        return default if default.present?
         raise Qa::IriTemplate::MissingParameter, "#{variable} is required, but missing" if required?
-        ''
+        default
```

Comment:
It is more 'right' to throw the exception for required variables before using the default.  Required variables do not have defaults.